app/controllers/BoxNumberController.scala

```diff
@@ -59,10 +59,10 @@ class BoxNumberController @Inject()(identity: IdentifierAction,
           updatedAnswers <- Future.fromTry(request.userAnswers.set(UnderpaymentReasonBoxNumberPage, value))
           _ <- sessionRepository.set(updatedAnswers)
         } yield {
-          appConfig.boxNumberItems.getOrElse(value, "notValid") match {
-            case "item" => Redirect(controllers.routes.ItemNumberController.onLoad())
-            case "entry" => Redirect(controllers.routes.BoxNumberController.onLoad())
-            case _ => BadRequest(view(formProvider(), backLink))
+          appConfig.boxNumberTypes.getOrElse(value, appConfig.invalidBox) match {
+            case box if(box.boxLevel.equals("item")) => Redirect(controllers.routes.ItemNumberController.onLoad())
+            case box if(box.boxLevel.equals("entry")) => Redirect(controllers.routes.UnderpaymentReasonAmendmentController.onLoad(box.boxNumber))
+            case box => BadRequest(view(formProvider(), backLink))
```

Comment:
The following removes the warnings in the file
```suggestion
            case box if box.boxLevel == "item" => Redirect(controllers.routes.ItemNumberController.onLoad())
            case box if box.boxLevel == "entry" => Redirect(controllers.routes.UnderpaymentReasonAmendmentController.onLoad(box.boxNumber))
            case _ => BadRequest(view(formProvider(), backLink))
```